src/Message/AbstractRequest.php

```diff
@@ -56,29 +56,34 @@ public function setGivingType($value)
     public function sendData($data)
     {
         $url = $this->getEndpoint();
-        $httpResponse = $this->httpClient->post(
+        $httpResponse = $this->httpClient->request(
+        'POST',
             $url,
-            null,
-            $data,
-            array(
-                'auth' => array($this->getPublicKey(), $this->getPrivateKey())
+            [],
+            json_encode(
+                array_merge(
+                    ['auth' => [$this->getPublicKey(), $this->getPrivateKey()]],
```

Comment:
The API documentation doesn't indicate that the authentication details can go into the request payload.

Could you please explain the reason for this change?

https://docs.tithe.ly/reference#charge-object